plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -650,19 +649,31 @@ hoistExpr var t = do
         (PIR.Def var' (PIR.mkVar var', PIR.Strict))
         mempty
 
-      t' <- maybeProfileRhs var' =<< addSpan (compileExpr t)
+      t' <- maybeProfileRhs var var' =<< addSpan (compileExpr t)
       -- See Note [Non-strict let-bindings]
       PIR.modifyTermDef lexName (const $ PIR.Def var' (t', PIR.NonStrict))
       pure $ PIR.mkVar var'
 
+-- 'GHC.Var' in argument is only for extracting srcspan and accurate name.
 maybeProfileRhs
-  :: (CompilingDefault uni fun m ann) => PLCVar uni -> PIRTerm uni fun -> m (PIRTerm uni fun)
-maybeProfileRhs var t = do
+  :: (CompilingDefault uni fun m ann)
+  => GHC.Var
+  -> PLCVar uni
+  -> PIRTerm uni fun
+  -> m (PIRTerm uni fun)
+maybeProfileRhs ghcVar var t = do
   CompileContext{ccOpts = compileOpts} <- ask
-  let ty = PLC._varDeclType var
-      varName = PLC._varDeclName var
-      displayName = T.pack $ PP.displayPlcSimple varName
-      isFunctionOrAbstraction = case ty of PLC.TyFun{} -> True; PLC.TyForall{} -> True; _ -> False
+  let
+    nameStr = GHC.occNameString $ GHC.occName $ GHC.varName $ ghcVar
+    displayName = T.pack $
+      case getVarSourceSpan ghcVar of
+        -- FIXME: Variables will miss span information when the module they are defined
+        -- in is loaded from cache instead of getting compiled.
```

Comment:
We have recently agreed to avoid leaving orphaned `TODO`, `FIXME` comments in the code. The best course of action would be:

- Create a github issue, such that a FIXME or TODO work can be a part of our dev process: triaged, assigned to a sprint, assigned to a person ("owned") etc.
- Optionally amend your `FIXME` comment with a link to the issue.

(The "triage" role that you assume this sprint includes work on removing such orphaned comments ;-) )

Reply:
done #7203